File: python/grains/grains.py
def on_square(integer_number):
  if integer_number < 1 or integer_number > 64:
    raise ValueError('Number must be between 1 & 64 (inclusive).')
  return 1 << integer_number - 1


def total_after(integer_number):
  if integer_number < 1 or integer_number > 64:
    raise ValueError('Number must be between 1 & 64 (inclusive).')
  return (1 << integer_number) - 1

File: kotlin/grains/src/main/kotlin/Grains.kt
import java.math.BigInteger

object Board {

    fun getGrainCountForSquare(number: Int): BigInteger {
        if (number < 1 || number > 64) {
            throw IllegalArgumentException("Only integers between 1 and 64 (inclusive) are allowed")
        }
        return BigInteger("1").shiftLeft(number - 1)
    }

    fun getTotalGrainCount(): BigInteger {
        return BigInteger("1").shiftLeft(64) - BigInteger("1")
    }
}
